File: ios/Classes/model/ActiveBeaconWires.swift
import CoreLocation

class ActiveBeaconWires {
    static func fromRegion(_ region: CLRegion, rssi: Int? = nil) -> ActiveBeaconWire? {
        guard let beaconRegion = region as? CLBeaconRegion else { return nil }
        
        return ActiveBeaconWire(
            id: beaconRegion.identifier,
            uuid: beaconRegion.uuid.uuidString,
            major: beaconRegion.major?.int64Value,
            minor: beaconRegion.minor?.int64Value,
            rssi: rssi != nil ? Int64(rssi!) : nil,
            triggers: [
                beaconRegion.notifyOnEntry ? .enter : nil,
                beaconRegion.notifyOnExit ? .exit : nil
            ].compactMap { $0 },
            androidSettings: nil
        )
    }
}

File: android/src/main/kotlin/com/chunkytofustudios/native_geofence/util/ActiveBeaconWires.kt
package com.chunkytofustudios.native_geofence.util

import com.chunkytofustudios.native_geofence.generated.ActiveBeaconWire
import com.chunkytofustudios.native_geofence.generated.BeaconWire

class ActiveBeaconWires {
    companion object {
        fun fromBeaconWire(e: BeaconWire): ActiveBeaconWire {
            return ActiveBeaconWire(
                e.id,
                e.uuid,
                e.major,
                e.minor,
                e.triggers,
                e.androidSettings
            )
        }
    }
}
